Mathlib/Combinatorics/Additive/FreimanHom.lean

```diff
@@ -289,6 +291,50 @@ lemma IsMulFreimanHom.inv (hf : IsMulFreimanHom n A B f) : IsMulFreimanHom n A B
 
 end DivisionCommMonoid
 
+namespace Fin
+variable {k m n : ℕ}
+
+private lemma aux (hm : m ≠ 0) (hkmn : m * k ≤ n) : k < n.succ :=
+  Nat.lt_succ_iff.2 $ le_trans (Nat.le_mul_of_pos_left _ hm.bot_lt) hkmn
+
+/-- **No wrap-around principle**.
+
+The first `k + 1` elements of `Fin (n + 1)` are `m`-Freiman isomorphic to the first `k + 1` elements
+of `ℕ` assuming there is no wrap-around. -/
+lemma isAddFreimanIso_Iic (hm : m ≠ 0) (hkmn : m * k ≤ n) :
+    IsAddFreimanIso m (Iic (k : Fin n.succ)) (Iic k) val where
```

Comment:
`n.succ` is no longer generated by `induction`, and it's also not really SNF. Is there a specific reason why you prefer it over `n+1`? Also elsewhere.